include/golioth/location.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
```

Comment:
Let's go ahead and add our ifdef C++ macros here to match other include files 👍🏻